File: golang/153_find_minimum_in_rotated_sorted_array.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func findMin(nums []int) int {
	l := 0
	h := len(nums) - 1
	if h < 0 {
		return 0
	}
	for (l + 1) < h {
		m := l + (h-l)/2
		if nums[m] > nums[l] && nums[m] > nums[h] {
			l = m
		} else {
			h = m
		}
	}
	if nums[l] < nums[h] {
		return nums[l]
	}
	return nums[h]
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	line := strings.Split(scanner.Text(), ",")
	var nums []int
	for i := 0; i < len(line); i++ {
		num, err := strconv.Atoi(line[i])
		if err == nil {
			nums = append(nums, num)
		}
	}
	fmt.Println(findMin(nums))
}

File: python/153_find_minimum_in_rotated_sorted_array.py
from typing import List

class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        h = len(nums) - 1
        if h < 0:
            return ''
        while l + 1 < h:
            m = l + (h - l) // 2
            if nums[m] > nums[l] and nums[m] > nums[h]:
                l = m
            else:
                h = m
        return min(nums[l], nums[h])

sol = Solution()
inp = [int(x) for x in input().split(",")]
print(sol.findMin(inp))